airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py

```diff
@@ -158,3 +159,12 @@ def availability_strategy(self) -> AbstractFileBasedAvailabilityStrategy:
     @property
     def name(self) -> str:
         return self.config.name
+
+    def get_cursor(self) -> Optional[Cursor]:
+        """
+        This is a temporary hack. Because file-based, declarative, and concurrent have _slightly_ different cursor implementations
+        the file-based cursor isn't compatible with the cursor-based iteration flow in core.py top-level CDK. By setting this to
+        None, we defer to the regular incremental checkpoint flow. Once all cursors are consolidated under a common interface
+        then this override can be removed.
```

Comment:
Actually after some thoughts around substreams, this is still a _slight_ hack, but needed for Python sources because determining if an HttpStream is a substream is very difficult. And allowing us to define a stream's cursor is easier than doing this inference.

I'll update this comment, but the idea remains that by default we'll return no cursor, and if a stream is an RFR substream, then we would override this on a per-source basis.

Reply:
we actually don't because it already implements these methods. In this case, a file based stream should be able to use a cursor based checkpoint reader where the cursor is probably instantiated to `v4/cursor.py:Cursor`